credentialdigger/cli/get_discoveries.py

```diff
@@ -0,0 +1,279 @@
+        csv_writer.writeheader()
+        csv_writer.writerows(discoveries)
+        csv_data = stringIO.getvalue()
+    except IndexError as error:
+        logger.error(error)
+    except Exception as exception:
+        logger.exception(exception)
+
+    return csv_data
+
+
+def export_csv(discoveries, client, save=False):
+    """ Export discoveries as a CSV file
+
+    Parameters
+    ----------
+        discoveries: list
+            List of discoveries from which to generate the CSV
+        client: `credentialdigger.Client`
+            Instance of the client from which we retrieve rules
+        save: bool
+            If True, we do not ask the user to enter a file path for the CSV
+            to be exported
+    """
+    # Check if --save is specified
+    if save is False:
+        path = ''
+        # Read the export path from the console's input
+        while path == '':
+            path = console.input('Path to export CSV:')
+    else:
+        # if --save argument is set, we use it as an export path
+        path = save
+
+    try:
+        with open(path, newline='', mode='w') as csv_file:
+            with console.status('[bold]Exporting the discoveries..'):
+                # Add the category to each discovery
+                assign_categories(client, discoveries)
+                data = discoveries_to_csv(discoveries)
+                csv_file.writelines(data)
+                console.print(
+                    '[bold][!] The discoveries have been exported \
+successfully.')
+    except Exception as e:
+        console.print(f'[red]{e}[/]')
+        try:
+            os.remove(path)
+        except OSError as osE:
+            console.print(f'[red]{osE}[/]')
+
+
+def assign_categories(client, discoveries):
+    """ Add category to each discovery
+
+    Parameters
+    ----------
+    client: `credentialdigger.Client`
+        Instance of the client from which we retrieve rules
+    discoveries: list
+        List of discoveries without assigned categories to them
+    """
+    rulesdict = client.get_rules()
+    for discovery in discoveries:
+        if discovery['rule_id']:
+            category = rulesdict[discovery['rule_id'] - 1]['category']
+            discovery['category'] = category
+        else:
+            discovery['category'] = '(rule deleted)'
+
+
+def filter_discoveries(discoveries, state):
+    """ Filter discoveries based on state
+
+    Parameters
+    ----------
+    discoveries: list
+        List of discoveries to be filtered
+    states: str | list
+            - str: if it equals 'all', then return all discoveries.
+                   return chosen state otherwise (i.e 'false_positive')
+            - list: return all the discoveries that have states contained
+                    in this list (i.e ['new', 'false_positive'])
+
+    Returns
+    -------
+    list
+        Filtered list of discoveries
+    """
+    if state == 'all':
```

Comment:
if the state is not specified we export all the discoveries. If the state is set, we export only that state.
Let's not give the possibility to set state=all because "all" never appears in the definition of state